fluss-server/src/main/java/org/apache/fluss/server/zk/ZooKeeperClient.java

```diff
@@ -1188,6 +1192,49 @@ public void insertConfigChangeNotification() throws Exception {
                         ZkData.ConfigEntityChangeNotificationSequenceZNode.encode());
     }
 
+    // --------------------------------------------------------------------------------------------
+    // Maintenance
+    // --------------------------------------------------------------------------------------------
+
+    public void registerServerTags(ServerTags newServerTags) throws Exception {
+        String path = ServerTagsZNode.path();
+        if (getOrEmpty(path).isPresent()) {
+            zkClient.setData().forPath(path, ServerTagsZNode.encode(newServerTags));
+        } else {
+            zkClient.create()
+                    .creatingParentsIfNeeded()
+                    .withMode(CreateMode.PERSISTENT)
+                    .forPath(path, ServerTagsZNode.encode(newServerTags));
+        }
+    }
+
+    public Optional<ServerTags> getServerTags() throws Exception {
+        String path = ServerTagsZNode.path();
+        return getOrEmpty(path).map(ServerTagsZNode::decode);
+    }
+
+    public void registerRebalancePlan(RebalancePlan rebalancePlan) throws Exception {
+        String path = RebalanceZNode.path();
+        if (getOrEmpty(path).isPresent()) {
+            zkClient.setData().forPath(path, RebalanceZNode.encode(rebalancePlan));
+        } else {
+            zkClient.create()
+                    .creatingParentsIfNeeded()
+                    .withMode(CreateMode.PERSISTENT)
+                    .forPath(path, RebalanceZNode.encode(rebalancePlan));
+        }
+    }
+
+    public Optional<RebalancePlan> getRebalancePlan() throws Exception {
+        String path = RebalanceZNode.path();
+        return getOrEmpty(path).map(RebalanceZNode::decode);
+    }
+
+    public void deleteRebalancePlan() throws Exception {
```

Comment:
I doubt whether we should delete the rebalance node on ZK. If we delete it once the rebalance successfully finished, we don't have the ability to respond users the status of the rebalance plan.

Maybe we can just soft delete the rebalance node by mark the rebalance plan is cancelled or failed or completed (the `RebalanceStatus`), so `listRebalanceProgress` still work after the rebalance finished, and can be overriden by next rebalance task.